Login crashes if .config/reliably/config.yaml is not present or lacks an auth object

**Describe the bug**

Login crashes if .config/reliably/config.yaml is not present or lacks an auth object

**To Reproduce**

```
How would you like to authenticate? Login with GitHub
DEBU[0000] Authorized callback url http://127.0.0.1:54108/login/with/github/authorized
DEBU[0000] open https://github.com/login/oauth/authorize?client_id=x&redirect_uri=x&response_type=code&scope=user%3Aemail&state=x
DEBU[0001] server handler: /login/with/github/authorized
DEBU[0001] server received query params map[code:[x] state:[x]]
DEBU[0001] server received code "x"
DEBU[0001] API call to /login/with/cli/github/authorized took 0.60002696 seconds
panic: assignment to entry in nil map

goroutine 1 [running]:
github.com/reliablyhq/cli/config.SetAuthInfo(0x28bf3cb, 0xc, 0xc00066ee10, 0x28, 0xc000119e30, 0x7, 0x7, 0x0)
	/home/runner/work/cli/cli/config/auth.go:73 +0x77
github.com/reliablyhq/cli/cmd/reliably/auth.loginRun(0xc0008d5a10, 0xffffffffffffff01, 0x5)
	/home/runner/work/cli/cli/cmd/reliably/auth/login.go:156 +0x4bd
github.com/reliablyhq/cli/cmd/reliably/auth.NewCmdLogin.func2(0xc0008c8b00, 0xc00011ad80, 0x0, 0x1, 0x0, 0x0)
	/home/runner/work/cli/cli/cmd/reliably/auth/login.go:80 +0xaf
github.com/spf13/cobra.(*Command).execute(0xc0008c8b00, 0xc00011ad60, 0x1, 0x1, 0xc0008c8b00, 0xc00011ad60)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:850 +0x472
github.com/spf13/cobra.(*Command).ExecuteC(0xc0008c8580, 0x5, 0x0, 0x0)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:958 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:895
github.com/reliablyhq/cli/cmd/reliably.Execute()
	/home/runner/work/cli/cli/cmd/reliably/root.go:143 +0x9e
main.main()
	/home/runner/work/cli/cli/main.go:21 +0x25
```

**Expected behavior**

Should log you in without breaking

**Additional context**

Solved by `mkdir -p ~/.config/reliably && echo 'auths: {}' > ~/.config/reliably/config.yaml && reliably auth login`

--- a/config/file.go
+++ b/config/file.go
@@ -1,7 +1,6 @@
 package config
 
 import (
-	"io/fs"
 	"os"
 
 	"github.com/mitchellh/go-homedir"
@@ -39,7 +38,7 @@ func writeConfigFile(data *Config) error {
 	}
 
 	p := resolveConfigFilePath()
-	if err := os.WriteFile(p, bytes, fs.ModeAppend); err != nil {
+	if err := os.WriteFile(p, bytes, 0600); err != nil {
 		return err
 	}
 	return nil

--- a/config/types.go
+++ b/config/types.go
@@ -19,5 +19,7 @@ type (
 
 // NewConfig initializes a config with empty/default values
 func NewConfig() *Config {
-	return &Config{}
+	return &Config{
+		AuthInfo: map[string]AuthInfo{},
+	}
 }

